mod_ci/controllers.py

```diff
@@ -181,7 +181,7 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
         config.get('SAMPLE_REPOSITORY', ''), 'vm_data', kvm_name, 'ci-tests')
     categories = Category.query.order_by(Category.id.desc()).all()
     commit_hash = GeneralData.query.filter(
-        GeneralData.key == 'last_commit').first().value
+        GeneralData.key == 'previous_commit').first().value
```

Comment:
So you're always comparing to the 2nd to last commit, even if it's a PR, which should be checked against the latest commit?